go/vt/etcdtopo/server_test.go

```diff
@@ -5,48 +5,127 @@
 package etcdtopo
 
 import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
 	"path"
 	"testing"
 	"time"
 
 	"golang.org/x/net/context"
 
-	"github.com/youtube/vitess/go/flagutil"
+	"github.com/youtube/vitess/go/testfiles"
 	"github.com/youtube/vitess/go/vt/topo"
 	"github.com/youtube/vitess/go/vt/topo/test"
 
 	topodatapb "github.com/youtube/vitess/go/vt/proto/topodata"
 )
 
-func newTestServer(t *testing.T, cells []string) *Server {
-	s := &Server{
-		_cells:    make(map[string]*cellClient),
-		newClient: newTestClient,
+// startEtcd starts an etcd subprocess, and waits for it to be ready.
+func startEtcd(t *testing.T) (*exec.Cmd, string, string) {
+	// Create a temporary directory.
+	dataDir, err := ioutil.TempDir("", "etcd")
+	if err != nil {
+		t.Fatalf("cannot create tempdir: %v", err)
 	}
 
-	// In tests, use cell name as the address.
-	globalAddrs = flagutil.StringListValue([]string{topo.GlobalCell})
-	c := s.getGlobal()
+	// Get our two ports to listen to.
+	port := testfiles.GoVtEtcdtopoPort
+	name := "vitess_unit_test"
+	clientAddr := fmt.Sprintf("http://localhost:%v", port)
+	peerAddr := fmt.Sprintf("http://localhost:%v", port+1)
+	initialCluster := fmt.Sprintf("%v=%v", name, peerAddr)
+
+	cmd := exec.Command("etcd",
```

Comment:
I'm fine with running it as process.

But have you considered to run it in process as well? That would probably be faster and maybe there would even be a way to block until the startup is done?

The downside would be that we have to vendor more Go code which is only used by tests. An alternative there would be to have two govendor configurations?

Nonetheless, I understand that it's simpler to run just the binary and I'm fine with this approach.